extensions/amp-auto-ads/0.1/attributes.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../../../src/log';
+import {startsWith} from '../../../src/string';
+
+/** @const */
+const TAG = 'amp-auto-ads';
+
+/**
+ * @const {!Object<string, boolean>}
+ */
+const NON_DATA_ATTRIBUTE_WHITELIST = {
+  'type': true,
+  'layout': true,
+};
+
+/**
+ * @param {!Object} configObj
+ * @return {!Object<string, string>}
+ */
+export function getAttributesFromConfigObj(configObj) {
+  if (!configObj['attributes']) {
+    return {};
+  }
+  if (typeof configObj['attributes'] != 'object') {
+    dev().warn(TAG, 'attributes property not an object');
```

Comment:
These should all be `user` warnings (user really means publisher). Dev logs are just for us, the AMP core devs.